parcels/particleset/particlesetsoa.py

```diff
@@ -173,7 +180,43 @@ def ArrayClass_init(self, *args, **kwargs):
             self.repeatkwargs = kwargs
 
         ngrids = fieldset.gridset.size if fieldset is not None else 1
-        self._collection = ParticleCollectionSOA(_pclass, lon=lon, lat=lat, depth=depth, time=time, lonlatdepth_dtype=lonlatdepth_dtype, pid_orig=pid_orig, partitions=partitions, ngrid=ngrids, **kwargs)
+        interaction_xy = None
+        interaction_z = None
+        self._dirty_neighbor = True
```

Comment:
Not sure 'dirty' is the best word here. Could confuse users? And should this also be a function argument in the signature?

Reply:
In CS, it's customary to to declare data arrangements (e.g. trees) as 'dirty' (here, it's a technical term) if the arrangement does not reflect the underlying data anymore and needs to be recomputed, i.e. the structure is 'dirty' (i.e. not up to date) and needs to be cleaned (i.e. 'updated'). In CS-lingo, this is a standard term for what it wants to tell.

Reply:
I have added some comments on its use.